### PR TITLE
fix: back button android, return home on tap

### LINKED ISSUE #57: Back Button closes App
On Android, while using the `Cupertino` theme, if the back button of the device is used instead of the back button of the app, the app closes. Don't think this is supposed to happen.

[VIdeo demo](https://imgur.com/a/i7xHM7d)

### PR DESCRIPTION
**Pull request by shreyas1599**
Fixes #57 and #54 

- I could not use a list as for some reason indexing with a list did not work. The state was always at `null`. Instead had to use five variables and reused them for all the platforms.
- Added a controller for the `CupertinoTabScaffold` as I need to get the value of `index` in `onWillPop`. I initially used a temporary variable and updated its value each time a tab bar item was clicked but because of using `await` in `onWillPop` it created trouble. And a controller inherently seems like a better option.

Let me know if there are any changes to be made. Thanks!

### CODE REVIEW

File: lib/home.dart [hunk 1]
```diff
@@ -174,23 +182,97 @@ class _HomeState extends State<Home> {
...
+            return tab1;
+          case 1:
+            return tab2;
+          case 2: 
+            return tab3;
+          case 3: 
+            return tab4;
+          case 4: 
+            return tab5;
+        }
```
Comment (pd4d10) [inline lib/home.dart:201]: Could we reuse the logic here for other platforms? For example, GitLab would never get into case 3 and 4.

Comment (shreyas1599) [reply lib/home.dart:201]: Do you mean like writing a single switch statement without checking the platform as for example gitlab would not reach 3,4?

Comment (pd4d10) [reply lib/home.dart:201]: Yeah. I'm not sure if there are other side effects in this way.

Comment (shreyas1599) [reply lib/home.dart]: Done.

**Pull request merged**